src/cargo/ops/tree/mod.rs

```diff
@@ -213,7 +213,13 @@ pub fn build_and_print(ws: &Workspace<'_>, opts: &TreeOptions) -> CargoResult<()
         })
         .collect::<CargoResult<Vec<PackageIdSpec>>>()?;
 
-    print(ws.config(), opts, root_indexes, &pkgs_to_prune, &graph)?;
+    if root_indexes.len() == 0 {
+        ws.config().shell().warn("nothing to print.\n\n\
+        To find dependencies that require specific features or target platforms, \
+        try use options `--all-features` or `--target all` first, and then narrow your search scope accordingly.")?;
```

Comment:
Could you help verify the needs of suggesting `--all-features`? I feel like I was wrong. It might already be checked [here](https://github.com/rust-lang/cargo/blob/26f4c036610a8dfeb368406d7299ba8da0e50c84/src/cargo/ops/tree/mod.rs#L189-L189) or elsewhere querying pkgid.

Reply:
Thanks!